tests/e2e/workloads/bdi/fixtures.py

```diff
@@ -0,0 +1,254 @@
+import pytest
+import os
+import logging
+from ocs_ci.utility.utils import run_cmd, clone_repo, exec_cmd
+from ocs_ci.ocs import machine, node, ocp, constants
+from ocs_ci.framework import config
+import tempfile
+from ocs_ci.utility import templating
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def clone_ibm_chart_project(request):
+    """
+    Clone IBM Chart project from Github
+    """
+    class_instance = request.node.cls
+
+    def finalizer():
+        """
+        Delete bdi temporary directory
+        """
+        log.info(f"Deleting directory {class_instance.bdi_dir}")
+        if os.path.isdir(class_instance.bdi_dir):
+            run_cmd(cmd="rm -rf " + class_instance.bdi_dir)
+
+    request.addfinalizer(finalizer)
+
+    log.info(f"Cloning chart from github into {class_instance.bdi_dir}")
+    clone_repo(url=class_instance.chart_git_url, location=class_instance.bdi_dir)
+
+
+@pytest.fixture()
+def create_machineset(request):
+    """
+    Creates a machineset with replica 3.
+    Additional worker nodes are needed in order to support the workload execution (3 default workers is not enough)
+    Relevant only for IPI deployment.
+    If we want to use UPI deployment, we need to create the additional workers during the post deployment precess
+    """
+    class_instance = request.node.cls
+
+    def finalizer():
+        """
+        Delete machineset
+        """
+        if class_instance.ms_name is not None and machine.check_machineset_exists(
+            machine_set=class_instance.ms_name
+        ):
+            log.info(f"Deleting machineset {class_instance.ms_name}")
+            machine.delete_custom_machineset(class_instance.ms_name)
+
+    if config.ENV_DATA["deployment_type"].lower() == "ipi":
+        log.info("Creating machineset")
+        class_instance.ms_name = machine.create_custom_machineset(
+            instance_type="m5.4xlarge", zone="a"
+        )
+        machine.wait_for_new_node_to_be_ready(class_instance.ms_name)
+        log.info(
+            f"Adding {class_instance.machine_set_replica} more nodes to machineset {class_instance.ms_name}"
+        )
+        node.add_new_node_and_label_it(
+            machineset_name=class_instance.ms_name,
+            num_nodes=class_instance.machine_set_replica,
+            mark_for_ocs_label=False,
+        )
+        machine.wait_for_new_node_to_be_ready(class_instance.ms_name)
+        request.addfinalizer(finalizer)
```

Comment:
should be in line 53, directly below the finalized function implementation

Reply:
Done.
Moved the finalizer to be added only if we're running on IPI

Reply:
The intention is to avoid having helper functions under tests/ hierarchy because in some cases, these functions are being called from places under ocs_ci/ and this is the opposite of the infra-tests importing logic. see https://github.com/red-hat-storage/ocs-ci/issues/408


Consider placing the functions in a new file under either of ocs_ci/helpers, ocs_ci/utility/workloads or ocs_ci/ocs

Reply:
Your comment meant to be for the helper.py I guess.
Moved the helpers file to ocs_ci/helpers/.
Done